relation/index.js

```diff
@@ -108,7 +111,39 @@ module.exports = yeoman.generators.Base.extend({
           }
           return m;
         },
-        validate: validateName
+        validate: function(value) {
+
+            var done = this.async();
+
+            //validfn returns false if relationship 
+            //has the same name with properties
+            var validfn = function(value, cb) {
+
+              //set flag isValid
+              var isValid = true;
+           
+              modelDef.properties(function(err, list) {
+                if(err) return done(err);
+                else {
+                  //check each property name in the list
+                  async.each(list, function(property, callback) {
+                    if(property.name === value) isValid = false;
+                    callback();
+                  }, function(err) {
+                    if(err) return done(err);
+                    cb(null, isValid);
+                  });
```

Comment:
No need to use async when the check is sync. You can simplify this code as follows:

```js
var conflict = list.some(function(property) { return property.name === value; });
cb(null, !conflict);
```